lib/scala/connected-lock-manager/src/main/scala/org/enso/lockmanager/client/ConnectedLockManager.scala

```diff
@@ -81,17 +83,17 @@ class ConnectedLockManager extends LockManager {
 
   private def sendRequestAndWaitForResponse(
     request: Runtime.ApiRequest
-  ): Runtime.ApiResponse = {
+  ): Try[Runtime.ApiResponse] = {
     val future = getEndpoint.sendRequest(request)
-    Await.result(future, Duration.Inf)
+    Try(Await.result(future, Duration(3, TimeUnit.SECONDS)))
   }
```

Comment:
This timeout is far far far too small.

I remember my confusion why I put the infinite timeout here but now I remember and it did have its merit after all.

The reason is - we are here trying to acquire a lock for library resources. If another process is in the middle of installing that library, the lock will need to wait the whole time until the library finishes installation. With bad internet connection, for some big libraries that we may have in the future, this may even take _hours_.

Well, I think we still need a timeout - the compiler getting stuck for hours because of a background job doing installation is no good thing either - if the installation is taking longer than expected - fine - we just need to cancel the current job. We just should tell the user clearly what happened.

So I suggest:
1. Let's make the timeout a bit larger, e.g. 30 or 60 seconds. This way a routine installation happening in the background will not crash the compiler but just make it wait a bit. I guess in many cases an installation _will_ take longer than 10s, so any timeout less than that is IMO insufficient.
2. I think ideally we should detect the `TimeoutException` in `ResourceManager` and there convert it into a more user friendly message - e.g. "Timed out when waiting to acquire a lock for library Foo.Bar, perhaps it is being modified in the background. Please wait until all background installation finish and restart the engine to try again.".

Reply:
I increased the timeout but I don't like the current API either. `withResource` and `withResources` should return a `Try` rather than throwing undeclared exceptions. Changing that will however trigger an avalanche of other changes that don't belong to this PR and should be addressed separately.